homework22/src/task1/LinkedListDouble.java

```diff
@@ -0,0 +1,217 @@
+package task1;
+
+/**
+ * Represents a linked list of double values. Implements few standard methods
+ * for linked list manipulation.
+ * 
+ * @author ajla.eltabari
+ *
+ */
+public class LinkedListDouble {
+	private Node head;
+
+	/**
+	 * Empty Constructor Sets start node to null (initialize empty linked list).
+	 */
+	public LinkedListDouble() {
+		this.head = null;
+	}
+
+	/**
+	 * Adds node to the linked list Checks if list is empty and adds start node,
+	 * or adds node at the end of the list if it is not empty
+	 */
+	public void add(double value) {
+		if (head == null) {
+			head = new Node(value);
+		} else {
+			Node last = getLastNode();
+			last.setNextNode(new Node(value));
+		}
+	}
+
+	/**
+	 * Returns last node in the linked list.
+	 */
+	private Node getLastNode() {
+		if (head == null) {
+			return null;
+		}
+
+		Node temp = head;
+		while (temp.getNextNode() != null) {
+			temp = temp.getNextNode();
+		}
+		return temp;
+	}
+
+	/**
+	 * Removes node from the linked list, by index. If provided index does not
+	 * exist, throws IndexOutOfBoundsException.
+	 * 
+	 * @param index
+	 */
+	public double remove(int index) throws IndexOutOfBoundsException {
+		if (size() == 0) {
+			System.out.println("Linked list is empty.");
+			return -1;
+		} else if (index >= size() || index < 0) {
+			throw new IndexOutOfBoundsException();
+		} else if (index == 0) {
+			head = head.getNextNode();
+		}
+		Node temp = head;
+		for (int i = 0; i < index; i++) {
+			temp = temp.getNextNode();
+		}
+
+		Node previous = getPreviousNode(temp);
+		previous.setNextNode(temp.getNextNode());
+		return temp.value;
+	}
+
+	/**
+	 * Returns current size of the linked list. Counts all nodes that have child
+	 * nodes.
+	 * 
+	 * @returns integer value, that represents size of the linked list
+	 */
+	public int size() {
+		int size = 0;
+		Node currentNode = head;
+		while (currentNode.next != null) {
+			currentNode = currentNode.next;
+			size++;
+		}
+		return size;
+	}
+
+	/**
+	 * Returns previous node of the provided node
+	 * 
+	 * @param node
+	 * @return
+	 */
+	public Node getPreviousNode(Node node) {
+		if (node == head) {
+			return null;
+		}
+
+		Node temp = head;
+		while (temp.getNextNode() != node) {
+			temp = temp.getNextNode();
+		}
+
+		return temp;
```

Comment:
In case node is never found, invalid result will be returned.

In case end is reached, NPE will be thrown since != is not allowed to have a null as an operand.